### PR TITLE
added tests to Torsion.jl file

### PR DESCRIPTION
**Pull request by FiroozehAga**

### CODE REVIEW

File: test/QuadForm/Quad/ZGenus.jl [hunk 1]
```diff
@@ -264,6 +264,17 @@
         q2, _ = normal_form(q2)
         q2 = Hecke.gram_matrix_quadratic(q2)
         @test q1 == q2
+<<<<<<< HEAD
```
Comment (thofma) [inline test/QuadForm/Quad/ZGenus.jl:267]: There are some merge artifacts here.

**Review by thofma — CHANGES_REQUESTED** [review]

File: test/QuadForm/Quad/ZGenus.jl (hunk 1 above)
Comment (FiroozehAga) [reply test/QuadForm/Quad/ZGenus.jl:267]: Oh, sorry I didn't realise it earlier. Will make changes and push them soon. Thanks

File: test/QuadForm/Torsion.jl [hunk 2]
```diff
@@ -50,19 +50,28 @@
   @test basis_matrix(Hecke.cover(Hecke.orthogonal_submodule_to(T, S)[1])) == matrix(QQ, 3, 3, [1//2,0,0,0,1//2,0,0,0,1//2])
 
   #checks if a TorQuadMod is degenerate
-  @test Hecke.isdegenerate(T) == false
+  @test Hecke.isdegenerate(T) == true
```
Comment (simonbrandhorst) [inline test/QuadForm/Torsion.jl:53]: false is correct. There is a bug in orthogonal_submodule_to

File: test/QuadForm/Torsion.jl [hunk 3]
```diff
@@ -50,19 +50,28 @@
...
+  t = Hecke.TorQuadMod(matrix(QQ,1,1,[1//27]))
+  d = sub(t, gens(t)*3)[1]
+  @test Hecke.isdegenerate(d) == true
 
   #test for rescaled torsion quadratic module
   @test Hecke.gram_matrix_quadratic(Hecke.rescale(T, -1)) == matrix(QQ, 3, 3, [7//4,0,0,0,7//4,0,0,0,7//4])
-  t = Hecke.TorQuadMod(QQ[1//3 0; 0 1//9])
+  t = Hecke.TorQuadMod(QQ[3//9 0; 0 1//9])
```
Comment (simonbrandhorst) [inline test/QuadForm/Torsion.jl:60]: This change does not seem to make sense. 1//3==3//9

**Review by simonbrandhorst — CHANGES_REQUESTED** [review]

File: test/QuadForm/Torsion.jl (hunk 3 above)
Comment (simonbrandhorst) [inline test/QuadForm/Torsion.jl:60]:
```suggestion
  t = Hecke.TorQuadMod(QQ[1//3 0; 0 1//9])
```

File: test/QuadForm/Torsion.jl (hunk 2 above)
Comment (simonbrandhorst) [inline test/QuadForm/Torsion.jl:53]:
```suggestion
  @test Hecke.isdegenerate(T) == false
```

File: src/QuadForm/Torsion.jl [hunk 4]
```diff
@@ -546,9 +546,10 @@ function orthogonal_submodule_to(T::TorQuadMod, S::TorQuadMod)
   # Element of the ambient module which pair in mZZ with cover(T)
   orthogonal =  m * integral
   # We have to make sure we get a submodule
-  Ortho = intersect(lattice(V, B), lattice(V, orthogonal))
-  ortho = Hecke.discriminant_group(Ortho)
```
Comment (simonbrandhorst) [inline src/QuadForm/Torsion.jl:550]: The error is here. It is not the correct way to create a submodule.

**Review by simonbrandhorst — APPROVED** [review]

**Comment by thofma** [comment]
Seems that the tests are failing: https://github.com/thofma/Hecke.jl/pull/258/checks?check_run_id=2343635480#step:6:214

File: test/QuadForm/Torsion.jl [hunk 5]
```diff
@@ -47,22 +47,44 @@
   L = Zlattice(matrix(ZZ, [[2,0,0],[0,2,0],[0,0,2]]))
   T = Hecke.discriminant_group(L)
   S, _ = sub(T, gens(T))
-  @test basis_matrix(Hecke.cover(Hecke.orthogonal_submodule_to(T, S)[1])) == matrix(QQ, 3, 3, [1//2,0,0,0,1//2,0,0,0,1//2])
+  @test basis_matrix(Hecke.cover(Hecke.orthogonal_submodule_to(T, S)[1])) == basis_matrix(L)
+  L1 = Zlattice(ZZ[1 0 0 0 0 0 0 0 0 0;
```
Comment (simonbrandhorst) [inline test/QuadForm/Torsion.jl:51]: You can replace this with `identity_matrix(ZZ,10)`

Comment (FiroozehAga) [reply test/QuadForm/Torsion.jl:51]: oh ok. I didn't know how to call the identity matrix. Will do that. Thank you:)

**Pull request merged**